test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -661,6 +661,30 @@ private void beforeWrite(String blobName) throws IOException {
                 }
             }
 
+            @Override
+            public void writeBlobAtomic(
+                OperationPurpose purpose,
+                String blobName,
+                InputStream inputStream,
+                long blobSize,
+                boolean failIfAlreadyExists
+            ) throws IOException {
```

Comment:
Nit: It seems to me that we can get rid of the other variant now since the default implementation from `BlobContainer` delegates it to this one.

Reply:
We cannot remove the other `writeBlobAtomic` a few lines below, because it also has logic to simulate failures. But we can make the other BytesReference-based variant call this InputStream-based variant. Did that.